source/game/system/map/MapdataCheckPoint.hh

```diff
@@ -0,0 +1,111 @@
+#pragma once
+
+#include <game/system/map/MapdataAccessorBase.hh>
+
+#include <egg/math/Vector.hh>
+
+namespace System {
+
+class MapdataCheckPoint;
+struct LinkedCheckpoint {
+    MapdataCheckPoint *checkpoint;
+    EGG::Vector2f p0diff;
+    EGG::Vector2f p1diff;
+    f32 distance;
+};
+
+class MapdataCheckPointAccessor;
+class MapdataCheckPoint {
```

Comment:
Regardless of when they're used, we prefer to put forward declarations before any other declarations
```suggestion
class MapdataCheckPoint;
class MapdataCheckPointAccessor;

struct LinkedCheckpoint {
    MapdataCheckPoint *checkpoint;
    EGG::Vector2f p0diff;
    EGG::Vector2f p1diff;
    f32 distance;
};

class MapdataCheckPoint {
```